### PR TITLE
Add stage channel create functions

### PR DESCRIPTION
**Pull request by NoComment1105**
These appeared to be missing, so i have added em

### CODE REVIEW

File: core/src/commonMain/kotlin/behavior/GuildBehavior.kt [hunk 1]
```diff
@@ -824,6 +825,29 @@ public suspend inline fun GuildBehavior.createNewsChannel(
...
 }
 
+/**
+ * Requests to create a new stage channel.
+ *
+ * @return The created [StageChannel].
+ *
+ * @throws [RestRequestException] if something went wrong during the request.
+ */
+public suspend inline fun CategoryBehavior.createStageChannel(
```
Comment (lukellmann) [inline core/src/commonMain/kotlin/behavior/GuildBehavior.kt:835]: should have `GuildBehavior` as the receiver

**Review by lukellmann — CHANGES_REQUESTED** [review]

File: core/src/commonMain/kotlin/behavior/GuildBehavior.kt (hunk 1 above)
Comment (NoComment1105) [reply core/src/commonMain/kotlin/behavior/GuildBehavior.kt:835]: 💀 Oml

Comment (NoComment1105) [reply core/src/commonMain/kotlin/behavior/GuildBehavior.kt]: Fixed

File: core/src/commonMain/kotlin/behavior/GuildBehavior.kt [hunk 2]
```diff
@@ -824,6 +824,29 @@ public suspend inline fun GuildBehavior.createNewsChannel(
...
+    name: String,
+    builder: StageChannelCreateBuilder.() -> Unit = {}
+): StageChannel {
+    contract {
+        callsInPlace(builder, EXACTLY_ONCE)
+    }
+    val response = kord.rest.guild.createStageChannel(id, name) {
+        builder()
+        parentId = id
+    }
```
Comment (lukellmann) [inline core/src/commonMain/kotlin/behavior/GuildBehavior.kt:844]: setting the `parentId` to the guild id is also wrong, just noticed after previous comment

**Review by lukellmann — CHANGES_REQUESTED** [review]

File: core/src/commonMain/kotlin/behavior/GuildBehavior.kt (hunk 2 above)
Comment (NoComment1105) [reply core/src/commonMain/kotlin/behavior/GuildBehavior.kt:844]: Ah yep, I see it

Comment (NoComment1105) [reply core/src/commonMain/kotlin/behavior/GuildBehavior.kt]: Fixed again

File: rest/src/commonMain/kotlin/builder/channel/StageChannelCreateBuilder.kt [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    override var reason: String? = null
+
+    private var _bitrate: OptionalInt = OptionalInt.Missing
+    public var bitrate: Int? by ::_bitrate.delegate()
+
+    private var _userLimit: OptionalInt = OptionalInt.Missing
+    public var userLimit: Int? by ::_userLimit.delegate()
+
+    private var _rateLimitPerUser: Optional<DurationInSeconds> = Optional.Missing()
+    public var rateLimitPerUser: DurationInSeconds? by ::_rateLimitPerUser.delegate()
```
Comment (lukellmann) [inline rest/src/commonMain/kotlin/builder/channel/StageChannelCreateBuilder.kt:29]: this can just be `Duration`, it's not serialized here

**Review by lukellmann — CHANGES_REQUESTED** [review]
looks good now, only one last thing

File: rest/src/commonMain/kotlin/builder/channel/StageChannelCreateBuilder.kt (hunk 3 above)
Comment (lukellmann) [reply rest/src/commonMain/kotlin/builder/channel/StageChannelCreateBuilder.kt]: i've pushed the changes myself

**Review by lukellmann — APPROVED** [review]
thanks :)

**Pull request merged**